ouroboros-network/src/Ouroboros/Network/BlockFetch/DeltaQ.hs

```diff
@@ -69,6 +70,27 @@ comparePeerGSV activePeers salt (a, a_p) (b, b_p) =
                  inboundGSV  = GSV g_in  _s_in  _v_in
                } = g_out + g_in
 
+-- | Order two PeerGSVs based on `g`.
+-- Like comparePeerGSV but doesn't take active status into account
+comparePeerGSV' :: forall peer.
+      ( Hashable peer
+      )
+      => Int
+      -> (PeerGSV, peer)
+      -> (PeerGSV, peer)
+      -> Ordering
+comparePeerGSV' salt (a, a_p) (b, b_p) =
+    let gs_a = gs a
+        gs_b = gs b in
+    if abs (gs_a - gs_b) < 0.05*gs_a
```

Comment:
What about `abc (gs_a - gs_b) < 0.05 * (max gs_a gs_b)` (or `min gs_a gs_b`).  If this condition is not symmetric `compareGSV'` might not satisfy `Ord` axioms.

Could you add a test which verifies `Ord` [axioms](https://hackage.haskell.org/package/base-4.16.0.0/docs/Prelude.html#t:Ord).

Reply:
You're right. This bug also affects `compareGSV`. I'll implement your fix above and add some tests.